### DISCUSSION TITLE
Import existing EIP

### DISCUSSION THREAD

**Discussion by HannesOberreiter** [aws/aws-cdk#31542] [Q&A]
Hi all,

I could not find any function to import an existing elastic IP address into my cdk project. Any help or is this not possible?

Thanks in advance!
Cheers
Hannes

**Comment by pahud** [comment]
What is your purpose to import that? Can you share more about your use case?

**Reply by HannesOberreiter** [reply]
Hi,

Thanks for your response. It’s nothing too complex— the IP is an optional property in my stack. If not provided, one will be automatically created.

This IP will be assigned specific tags based on other resources in the Stack, used in a UserData script for an EC2 instance, and then the public IP is exported as a CfnOutput to be accessed by another stack.

The only issue I haven’t figured out yet is how to tag the resource if the IP is passed as a prop, but it's not a big deal. I can hardcode it if needed.

Overall, it’s not a critical issue, just more of a curiosity.

Cheers,
Hannes
